test/system/src/test/java/io/pravega/test/system/MultiReaderWriterTest.java

```diff
@@ -0,0 +1,267 @@
+    private static final int NUM_READERS = 20;
+    private static final long NUM_EVENTS = 20000;
+    private AtomicBoolean stopReadFlag;
+    private AtomicLong eventData;
+    private AtomicLong eventReadCount;
+    private ConcurrentLinkedQueue<Long> eventsReadFromPravega;
+    private Service controllerInstance = null;
+    private Service segmentStoreInstance = null;
+
+    @Environment
+    public static void initialize() throws InterruptedException, MarathonException, URISyntaxException {
+
+        //1. Start 1 instance of zookeeper
+        Service zkService = new ZookeeperService("zookeeper");
+        if (!zkService.isRunning()) {
+            zkService.start(true);
+        }
+        List<URI> zkUris = zkService.getServiceDetails();
+        log.debug("Zookeeper service details: {}", zkUris);
+        //get the zk ip details and pass it to bk, host, controller
+        URI zkUri = zkUris.get(0);
+
+        //2. Start 3 bookies
+        Service bkService = new BookkeeperService("bookkeeper", zkUri);
+        if (!bkService.isRunning()) {
+            bkService.start(true);
+        }
+        List<URI> bkUris = bkService.getServiceDetails();
+        log.debug("Bookkeeper service details: {}", bkUris);
+
+        //3. start 2 instances of pravega controller
+        Service conService = new PravegaControllerService("controller", zkUri, 2, 0.1, 700.0);
+        if (!conService.isRunning()) {
+            conService.start(true);
+        }
+        List<URI> conUris = conService.getServiceDetails();
+        log.debug("Pravega Controller service  details: {}", conUris);
+
+        //4.start 2 instances of pravega segmentstore
+        Service segService = new PravegaSegmentStoreService("segmentstore", zkUri, conUris.get(0), 2, 0.1, 1000.0);
+        if (!segService.isRunning()) {
+            segService.start(true);
+        }
+        List<URI> segUris = segService.getServiceDetails();
+        log.debug("Pravega segmentstore service  details: {}", segUris);
+    }
+
+    @Before
+    public void setup() {
+
+        //1. Start 1 instance of zookeeper
+        Service zkService = new ZookeeperService("zookeeper");
+        if (!zkService.isRunning()) {
+            zkService.start(true);
+        }
+        List<URI> zkUris = zkService.getServiceDetails();
+        log.debug("Zookeeper service details: {}", zkUris);
+        //get the zk ip details and pass it to bk, host, controller
+        URI zkUri = zkUris.get(0);
+
+        // Verify controller is running.
+        controllerInstance = new PravegaControllerService("controller",  zkUri, 2, 0.1, 700.0);
+        Assert.assertTrue(controllerInstance.isRunning());
+        List<URI> conURIs = controllerInstance.getServiceDetails();
+        log.info("Pravega Controller service instance details: {}", conURIs);
+
+        // Verify segment stores is running.
+        segmentStoreInstance = new PravegaSegmentStoreService("segmentstore", zkUri, conURIs.get(0), 2, 0.1, 1000.0);
```

Comment:
`getControllerURI()` returns only 1 URI of controller.
Here, I have 2 instances.